[docs] best-practices add use variable with template example. (#27316)
mistercrunch/Airflow: docs/apache-airflow/best-practices.rst

    {{ var.json.<variable_name> }}

> Ensure use variable with template in operator, not get it in top level code.

Reviewer: ```suggestion
Make sure to use variable with template in operator, not in the top level code.
```

Bad example:

.. code-block:: python

    from airflow.models import Variable

>     foo_var = Variable.get("foo")

Reviewer: ```suggestion
    foo_var = Variable.get("foo")  # DON'T DO THAT
```

    bash_use_variable_bad = BashOperator(
        task_id="bash_use_variable_bad", bash_command="echo variable foo=${foo_env}", env={"foo_env": foo_var}
    )

Good example:

.. code-block:: python

    bash_use_variable_good = BashOperator(
        task_id="bash_use_variable_good",
        bash_command="echo variable foo=${foo_env}",
        env={"foo_env": "{{ var.value.get('foo') }}"},
    )

> 

Reviewer: Would be great to add also a bad example where variable is also (wrongly) used as parameter:
```
bash_command="echo variable foo=${Variable.get('foo')}"  # DON'T DO THAT
env={"foo_env": Variable.get('foo')}, # DON'T DO THAT
```

And another example, where we show how you CAN use Variable.get in @task - decorated code.

```
@task
def my_task():
     var = Variable.get('foo')   # this is fine
     print(var)

Author: ? `bash_command="echo variable foo=${Variable.get('foo')}"` is not a effective syntax.
Are you want to say `bash_command=f"echo variable foo={Variable.get('foo')}"` ?
